onnxmltools/convert/sklearn/operator_converters/NaiveBayes.py

```diff
@@ -137,58 +169,51 @@ def convert_sklearn_naive_bayes(scope, operator, container):
                               feature_log_prob.shape, feature_log_prob.flatten())
     container.add_initializer(classes_name, class_type, classes.shape, classes)
+    container.add_initializer(class_log_prior_name, onnx_proto.TensorProto.FLOAT,
+                              class_log_prior.shape, class_log_prior.flatten())
 
     if operator.type == 'SklearnMultinomialNB':
-        container.add_initializer(class_log_prior_name, onnx_proto.TensorProto.FLOAT,
-                                  class_log_prior.shape, class_log_prior.flatten())
         matmul_result_name = scope.get_unique_variable_name('matmul_result')
+        shape_result_name = scope.get_unique_variable_name('shape_result')
+        reshape_result_name = scope.get_unique_variable_name('reshape_result')
 
         container.add_node('MatMul', [operator.inputs[0].full_name, feature_log_prob_name],
                            matmul_result_name, name=scope.get_unique_operator_name('MatMul'))
         # Cast is required here as Sum op doesn't work with Float64
-        container.add_node('Cast', matmul_result_name, cast_result_name,
-                           to=onnx_proto.TensorProto.FLOAT, op_version=7)
-                           
-        shape_result_name = scope.get_unique_variable_name('shape_result')
+        apply_cast(scope, matmul_result_name, cast_result_name, container,
+                   to=onnx_proto.TensorProto.FLOAT)
         container.add_node('Shape', class_log_prior_name, shape_result_name)
-        reshape_result_name = scope.get_unique_variable_name('reshape_result')
         container.add_node('Reshape', [cast_result_name, shape_result_name], reshape_result_name)
-        
         container.add_node('Sum', [reshape_result_name, class_log_prior_name],
                            sum_result_name, name=scope.get_unique_operator_name('Sum'))
     else:
-        container.add_initializer(class_log_prior_name, onnx_proto.TensorProto.FLOAT,
-                                  class_log_prior.shape, class_log_prior.flatten())
         constant_name = scope.get_unique_variable_name('constant')
         exp_result_name = scope.get_unique_variable_name('exp_result')
         sub_result_name = scope.get_unique_variable_name('sub_result')
         neg_prob_name = scope.get_unique_variable_name('neg_prob')
         sum_neg_prob_name = scope.get_unique_variable_name('sum_neg_prob')
-        inp_neg_prob_prod_name = scope.get_unique_variable_name('inp_neg_prob_prod')
         difference_matrix_name = scope.get_unique_variable_name('difference_matrix')
+        dot_prod_name = scope.get_unique_variable_name('dot_prod')
+        partial_sum_result_name = scope.get_unique_variable_name('partial_sum_result')
 
         container.add_initializer(constant_name, onnx_proto.TensorProto.FLOAT,
                                   [], [1.0])
-
         input_name = operator.inputs[0].full_name
 
         if nb.binarize is not None:
```

Comment:
Could nb.binarize be false?

Reply:
sklearn's code uses the same test:
https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/naive_bayes.py#L936

Reply:
Ah. I didn't know this. Thanks you @xadupre. Let me close this comment.